apps/backend/src/deposits/cash-deposit.service.ts

```diff
@@ -113,27 +110,50 @@ export class CashDepositService {
    *  pending or in progress. This is used to find the dates that need to be reconciled.
    */
   public async depositDates(event: ReconciliationEvent): Promise<string[]> {
-    const {
-      date,
-      program,
-      location: { pt_location_id }
-    } = event;
+    const { fiscal_close_date, fiscal_start_date, program, location } = event;
 
     const dates = await this.cashDepositRepo.find({
       select: { deposit_date: true },
       where: {
-        pt_location_id,
+        pt_location_id: location.pt_location_id,
         metadata: { program },
-        deposit_date: LessThanOrEqual(date)
+        deposit_date: Raw(
+          (alias) =>
+            `${alias} <= :fiscal_close_date and ${alias} > :fiscal_start_date `,
+          {
+            fiscal_close_date,
+            fiscal_start_date
+          }
+        )
       },
       order: {
-        deposit_date: 'ASC'
+        deposit_date: 'DESC'
       }
     });
 
     return Array.from(new Set(dates.map((item) => item.deposit_date)));
   }
+  /**
+   * @param deposits: CashDepositEntity[]
+   * @param status: MatchStatus
+   * @returns CashDepositEntity
+   */
+  async updateDeposits(
+    deposits: CashDepositEntity[],
+    status: MatchStatus
+  ): Promise<CashDepositEntity[]> {
+    this.appLogger.log(
+      `UPDATED: ${deposits.length} CASH DEPOSITS to ${status.toUpperCase()}`,
+      CashDepositService.name
+    );
 
+    return await Promise.all(
+      deposits.map(
+        async (deposit) =>
+          await this.updateDepositStatus({ ...deposit, status })
+      )
```

Comment:
I've moved the mapping for status updates out of cash reconciliation service in order to clean it up a bit.